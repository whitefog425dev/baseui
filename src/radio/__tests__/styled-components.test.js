/*
Copyright (c) 2018 Uber Technologies, Inc.

This source code is licensed under the MIT license found in the
LICENSE file in the root directory of this source tree.
*/
// @flow
import React from 'react';
import {shallow} from 'enzyme';
import {
  StyledRoot,
  StyledLabel,
  StyledRadioMarkInner,
  StyledRadioMarkOuter,
  StyledInput,
  StyledRadioGroupRoot,
} from '../index.js';

describe('RadioGroup styled components', () => {
  describe('StyledLabel', () => {
    test.each([[''], ['disabled'], ['$isError']])('', prop => {
      const props = {};
      props[prop] = true;
      const component = shallow(
        <StyledLabel {...props}>
          <div />
        </StyledLabel>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'has correct styles when ' + prop,
      );
    });
  });

  describe('StyledLabel', function() {
    test('Label', () => {
      const component = shallow(
        <StyledLabel>
          <div />
        </StyledLabel>,
      );
      ['left', 'right', 'top', 'bottom'].forEach(labelPlacement => {
        component.setProps({
          $labelPlacement: labelPlacement,
        });
        expect(component.instance().getStyles()).toMatchSnapshot(
          'StyledLabel has correct styles when set to ' + labelPlacement,
        );
      });
    });
  });

  describe('StyledRadioGroupRoot', function() {
    test('RadioGroupRoot', () => {
      const component = shallow(
        <StyledRadioGroupRoot>
          <div />
        </StyledRadioGroupRoot>,
      );
      ['horizontal', 'vertical'].forEach(align => {
        component.setProps({
          $align: align,
        });
        expect(component.instance().getStyles()).toMatchSnapshot(
          'StyledRadioGroupRoot has correct styles when set to ' + align,
        );
      });
    });
  });

  describe('StyledRadioMarkInner', () => {
    test.each([[''], ['disabled'], ['$isFocused'], ['checked'], ['$isError']])(
      '',
      prop => {
        const props = {};
        props[prop] = true;
        const component = shallow(
          <StyledRadioMarkInner {...props}>
            <div />
          </StyledRadioMarkInner>,
        );
        expect(component.instance().getStyles()).toMatchSnapshot(
          'has correct styles when ' + prop,
        );
      },
    );
  });

  describe('StyledRadioMarkOuter', function() {
    test('', () => {
      const component = shallow(
        <StyledRadioMarkOuter>
          <div />
        </StyledRadioMarkOuter>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledRadioMarkOuter has correct styles',
      );
    });
  });

  describe('StyledInput', function() {
    test('', () => {
      const component = shallow(
        <StyledInput>
          <div />
        </StyledInput>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledInput has correct styles',
      );
    });
  });

  describe('StyledRoot', function() {
    test('', () => {
      const component = shallow(
        <StyledRoot>
          <div />
        </StyledRoot>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'StyledRoot has correct styles',
      );
    });

    test.each([['top'], ['bottom'], ['left'], ['right']])('', prop => {
      const props = {
        $labelPlacement: prop,
      };
      const component = shallow(
        <StyledRoot {...props}>
          <div />
        </StyledRoot>,
      );
      expect(component.instance().getStyles()).toMatchSnapshot(
        'has correct styles when label is placed' + prop,
      );
    });
  });
});
