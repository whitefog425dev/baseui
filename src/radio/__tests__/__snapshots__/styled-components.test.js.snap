// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup styled components StyledInput : StyledInput has correct styles 1`] = `
Object {
  "margin": 0,
  "opacity": 0,
  "overflow": "hidden",
  "padding": 0,
  "width": 0,
}
`;

exports[`RadioGroup styled components StyledLabel : has correct styles when  1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "padding": "$theme.sizing.scale200",
  "verticalAlign": "middle",
}
`;

exports[`RadioGroup styled components StyledLabel : has correct styles when $isError 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "padding": "$theme.sizing.scale200",
  "verticalAlign": "middle",
}
`;

exports[`RadioGroup styled components StyledLabel : has correct styles when disabled 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "padding": "$theme.sizing.scale200",
  "verticalAlign": "middle",
}
`;

exports[`RadioGroup styled components StyledLabel Label: StyledLabel has correct styles when set to bottom 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingBottom": "$theme.sizing.scale200",
  "verticalAlign": "middle",
}
`;

exports[`RadioGroup styled components StyledLabel Label: StyledLabel has correct styles when set to left 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingLeft": "$theme.sizing.scale200",
  "verticalAlign": "middle",
}
`;

exports[`RadioGroup styled components StyledLabel Label: StyledLabel has correct styles when set to right 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingRight": "$theme.sizing.scale200",
  "verticalAlign": "middle",
}
`;

exports[`RadioGroup styled components StyledLabel Label: StyledLabel has correct styles when set to top 1`] = `
Object {
  "color": "$theme.colors.mono1000",
  "fontFamily": "$theme.typography.font400.fontFamily",
  "fontSize": "$theme.typography.font400.fontSize",
  "fontWeight": "$theme.typography.font400.fontWeight",
  "lineHeight": "$theme.typography.font400.lineHeight",
  "paddingTop": "$theme.sizing.scale200",
  "verticalAlign": "middle",
}
`;

exports[`RadioGroup styled components StyledRadioGroupRoot RadioGroupRoot: StyledRadioGroupRoot has correct styles when set to horizontal 1`] = `
Object {
  "alignItems": "center",
  "cursor": "pointer",
  "display": "flex",
  "flexDirection": "row",
}
`;

exports[`RadioGroup styled components StyledRadioGroupRoot RadioGroupRoot: StyledRadioGroupRoot has correct styles when set to vertical 1`] = `
Object {
  "alignItems": "flex-start",
  "cursor": "pointer",
  "display": "flex",
  "flexDirection": "column",
}
`;

exports[`RadioGroup styled components StyledRadioMark : has correct styles when  1`] = `
Object {
  ":active": Object {
    "backgroundColor": "$theme.colors.mono400",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono400",
  },
  "backgroundColor": "transparent",
  "backgroundImage": null,
  "backgroundPosition": "center",
  "backgroundRepeat": "no-repeat",
  "borderColor": "$theme.colors.mono700",
  "borderRadius": "50%",
  "borderStyle": "solid",
  "borderWidth": "$theme.sizing.scale0",
  "display": "inline-block",
  "flex": "0 0 auto",
  "height": "$theme.sizing.scale600",
  "marginBottom": "$theme.sizing.scale200",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "marginTop": "$theme.sizing.scale200",
  "transitionDuration": "$theme.animation.timing100",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "verticalAlign": "middle",
  "width": "$theme.sizing.scale600",
}
`;

exports[`RadioGroup styled components StyledRadioMark : has correct styles when $isError 1`] = `
Object {
  ":active": Object {
    "backgroundColor": null,
  },
  ":hover": Object {
    "backgroundColor": null,
  },
  "backgroundColor": "transparent",
  "backgroundImage": null,
  "backgroundPosition": "center",
  "backgroundRepeat": "no-repeat",
  "borderColor": "$theme.colors.negative400",
  "borderRadius": "50%",
  "borderStyle": "solid",
  "borderWidth": "$theme.sizing.scale0",
  "display": "inline-block",
  "flex": "0 0 auto",
  "height": "$theme.sizing.scale600",
  "marginBottom": "$theme.sizing.scale200",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "marginTop": "$theme.sizing.scale200",
  "transitionDuration": "$theme.animation.timing100",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "verticalAlign": "middle",
  "width": "$theme.sizing.scale600",
}
`;

exports[`RadioGroup styled components StyledRadioMark : has correct styles when $isFocused 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "$theme.colors.mono500",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono500",
  },
  "backgroundColor": "transparent",
  "backgroundImage": null,
  "backgroundPosition": "center",
  "backgroundRepeat": "no-repeat",
  "borderColor": "$theme.colors.mono700",
  "borderRadius": "50%",
  "borderStyle": "solid",
  "borderWidth": "$theme.sizing.scale0",
  "display": "inline-block",
  "flex": "0 0 auto",
  "height": "$theme.sizing.scale600",
  "marginBottom": "$theme.sizing.scale200",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "marginTop": "$theme.sizing.scale200",
  "transitionDuration": "$theme.animation.timing100",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "verticalAlign": "middle",
  "width": "$theme.sizing.scale600",
}
`;

exports[`RadioGroup styled components StyledRadioMark : has correct styles when checked 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "$theme.colors.mono400",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono400",
  },
  "backgroundColor": "transparent",
  "backgroundImage": null,
  "backgroundPosition": "center",
  "backgroundRepeat": "no-repeat",
  "borderColor": "$theme.colors.mono700",
  "borderRadius": "50%",
  "borderStyle": "solid",
  "borderWidth": "$theme.sizing.scale0",
  "display": "inline-block",
  "flex": "0 0 auto",
  "height": "$theme.sizing.scale600",
  "marginBottom": "$theme.sizing.scale200",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "marginTop": "$theme.sizing.scale200",
  "transitionDuration": "$theme.animation.timing100",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "verticalAlign": "middle",
  "width": "$theme.sizing.scale600",
}
`;

exports[`RadioGroup styled components StyledRadioMark : has correct styles when disabled 1`] = `
Object {
  ":active": Object {
    "backgroundColor": "$theme.colors.mono400",
  },
  ":hover": Object {
    "backgroundColor": "$theme.colors.mono400",
  },
  "backgroundColor": "transparent",
  "backgroundImage": null,
  "backgroundPosition": "center",
  "backgroundRepeat": "no-repeat",
  "borderColor": "$theme.colors.mono700",
  "borderRadius": "50%",
  "borderStyle": "solid",
  "borderWidth": "$theme.sizing.scale0",
  "display": "inline-block",
  "flex": "0 0 auto",
  "height": "$theme.sizing.scale600",
  "marginBottom": "$theme.sizing.scale200",
  "marginLeft": "$theme.sizing.scale200",
  "marginRight": "$theme.sizing.scale200",
  "marginTop": "$theme.sizing.scale200",
  "transitionDuration": "$theme.animation.timing100",
  "transitionTimingFunction": "$theme.animation.easeOutCurve",
  "verticalAlign": "middle",
  "width": "$theme.sizing.scale600",
}
`;

exports[`RadioGroup styled components StyledRoot : StyledRoot has correct styles 1`] = `
Object {
  "alignItems": "center",
  "cursor": "pointer",
  "display": "flex",
  "flexDirection": "row",
}
`;

exports[`RadioGroup styled components StyledRoot : has correct styles when label is placedbottom 1`] = `
Object {
  "alignItems": "center",
  "cursor": "pointer",
  "display": "flex",
  "flexDirection": "column",
}
`;

exports[`RadioGroup styled components StyledRoot : has correct styles when label is placedleft 1`] = `
Object {
  "alignItems": "center",
  "cursor": "pointer",
  "display": "flex",
  "flexDirection": "row",
}
`;

exports[`RadioGroup styled components StyledRoot : has correct styles when label is placedright 1`] = `
Object {
  "alignItems": "center",
  "cursor": "pointer",
  "display": "flex",
  "flexDirection": "row",
}
`;

exports[`RadioGroup styled components StyledRoot : has correct styles when label is placedtop 1`] = `
Object {
  "alignItems": "center",
  "cursor": "pointer",
  "display": "flex",
  "flexDirection": "column",
}
`;
